### ISSUE TITLE
Validation trigger indicator

### ISSUE THREAD

**Issue by insin** [erikras/redux-form#81]
I have a validation use case I'm currently using a separate global `messages` duck in my `onSubmit` handler for:

* Search filter form
  * All individual fields are optional but the user is required to populate at least one before submitting.
  * Only display the global "Please provide some search criteria." form error on submit.

Now that redux-form has global errors (yay!), I'd like to move this validation into my form instead of manually handling it elsewhere. This would also allow me to take advantage of onBlur and onChange to clear the error once the user populates a field. This is synchronous validation. but I can't currently put it in `validate()` because it runs when initialising the form (so it starts with the global error displayed, as the form is empty).

Is it possible to pass something to validation functions to indicate what triggered validation? Two specific cases I'm interested in treating differently are are initial form display (skipping validation for forms which are initialised in an invalid state which would normally trigger a global error) and form submission (only performing certain synchronous validation on submit).

**Comment by erikras** [comment]
This is the whole motivation for the `touched` flags, to only show the validation error when a field has been touched, but you're referring to the whole form, so perhaps a global `touched` variable might be of some use for this. What do you think?

**Comment by andrewmclagan** [comment]
Yes, a global touched variable would be great!

Think of this use-case: 

````
<form>
	<input type="text" {...fieldOne} />
	<input type="text" {...fieldTwo} />
	<input type="text" {...fieldThree} />

	{ (invalid && anyTouched) && <p>Sorry, looks like you made some mistakes. Check the form and try again</p>}

	<button disabled={invalid || submitting}>Submit</button>
</form>
````

**Comment by erikras** [comment]
Ah, an `anyTouched` flag. `v6` has an `allTouched` flag internally that is set when a submission fails.

I guess an `anyTouched` flag wouldn't require much rerendering, as it will literally only ever cause one rerender. Not a bad idea at all!

**Comment by erikras** [comment]
@andrewmclagan `anyTouched` implemented in [`v6.0.0-alpha.5`](https://github.com/erikras/redux-form/releases/tag/v6.0.0-alpha.5)

**Comment by andrewmclagan** [comment]
:-)
